bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/WifiStationHandler.java

```diff
@@ -90,4 +98,26 @@ private void updateRateChannel(String channel, long rate) {
     private int toQoS(int rssi) {
         return rssi > -50 ? 4 : rssi > -60 ? 3 : rssi > -70 ? 2 : rssi > -85 ? 1 : 0;
     }
+
+    @Override
+    protected LanHost getLanHost() throws FreeboxException {
+        try {
+            return super.getLanHost();
+        } catch (FreeboxException e) {
+            HostName identifier = getConfigAs(WifiHostConfiguration.class).getIdentifier();
+            if (identifier != null) {
+                cancelPushSubscription();
+                Optional<LanHost> lanHost = getManager(LanBrowserManager.class).getHost(identifier);
+                return lanHost.map(host -> {
+                    Configuration thingConfig = editConfiguration();
+                    thingConfig.put(Thing.PROPERTY_MAC_ADDRESS, host.getMac().toColonDelimitedString());
+                    updateConfiguration(thingConfig);
```

Comment:
Just checking, tested with textual configuration also?

Reply:
Yes, it works but thing configuration change is volatile - not saved in .things file for sure. This is not really an issue as resolution will be redone based on mDNS name on next restart